### ISSUE TITLE
Applying authentication config to replication

### ISSUE THREAD

**Issue by derekdon** [couchbaselabs/nativescript-couchbase#35]
I'm trying to port my ionic app and I've hit a snag. I was using PouchDB with the session information returned from my custom authentication api. The endpoint creates the sync gateway session for the user. The response to the app might look like this...

```
{
     token: 'somelongapitoken',
     session: { 
          session_id: 'somelongid',
          expires: '2016-09-26T00:41:11.542304529Z',
          cookie_name: 'SyncGatewaySession',
          cookie_opts: { 
             expires: Mon Sep 26 2016 01:41:11 GMT+0100 (BST),
             httpOnly: false
          }
      },
     user: { 
          name: 'derek... etc'
     }
} 
```

I'm not sure how to apply this information the the subsequent calls to enable replication, any suggestions?

I'm using version 1.0.11 of nativescript-couchbase and 2.3.0 of tns.

I thought there might be something like this:

```
push.setAuthenticator(auth);
pull.setAuthenticator(auth);
```

or 

```
pull.setCookie(session.get("cookie_name"), session.get("session_id"), "/", date, false, false);
```

I should point out that I've never used Couchbase Lite before, generally just PouchDB > Sync Gateway > Couchbase Server in the context of hybrid webview apps. Also I'm pretty new to NativeScript.

**Comment by derekdon** [comment]
In my case I think I need to use the setCookie method on the replication object. It looks like this fork https://github.com/martingasser/nativescript-couchbase may have alreay exposed the setCookie method, so I'll try that.

**Comment by nraboy** [comment]
I hadn't added auth to the plugin yet, but it shouldn't be difficult.  @hodgreeley want to have a look?

Or maybe @martingasser wants to submit a PR to get any changes he made included in the upstream.

**Comment by HodGreeley** [comment]
I'll be taking a look at some point, but I don't know when I can get to it.

@derekdon did @martingasser's repo work for you?

**Comment by derekdon** [comment]
@HodGreeley no I haven't been able to get this functionality working yet as there's been some integration issues. See https://github.com/couchbaselabs/nativescript-couchbase/pull/37 for reference.

**Comment by feedyourmind** [comment]
the cookie configuration pull-request has been merged, but the code still hasn't been published to npm,
is this going to happen soon? We really need this feature, otherwise replication can not work in our case,  which means we won't be able to use couchbase at all.

**Comment by nraboy** [comment]
NPM lists 1.0.12:

https://www.npmjs.com/package/nativescript-couchbase

Checked and saw these lines in one of the files:

![screen shot 2016-11-17 at 12 10 39 pm](https://cloud.githubusercontent.com/assets/983958/20405704/29d79128-acbf-11e6-9dcf-289f3547acc2.png)

@martingasser was the author of that change.  Martin, care to comment?

It is under **couchbaselabs** so it is a community effort.  It is nothing more than a JavaScript wrapper to the native SDKs, so as long as the native SDKs work, so will this plugin.  Lack of activity does not mean abandoned.  It isn't one of those high maintenance projects that needs to be nursed every hour.  It works and it works well.

This user asked the same thing:

https://github.com/couchbaselabs/nativescript-couchbase/issues/44

**Comment by martingasser** [comment]
This was already fixed in 8f3b8de504725578466ce86a7b1cd45b4bb41901

@nraboy, can you publish an update to NPM?

**Comment by nraboy** [comment]
I just published 1.0.13 to NPM.  There shouldn't be any issues now.  Not sure why it didn't take previously.
